astropy/cosmology/io/html.py

```diff
@@ -0,0 +1,161 @@
+        the cosmology instance. The class also provides default parameter
+        values, filling in any non-mandatory arguments missing in 'table'.
+    **kwargs : Any
+        Passed to :attr:`astropy.table.QTable.read`.
+        ``format`` is set to 'ascii.html', regardless of input.
+
+    Returns
+    -------
+    |Cosmology| subclass instance
+    """
+    kwargs["format"] = "ascii.html"  # force the format
+    # Reading is handled by `QTable`.
+    with u.add_enabled_units(cu):  # (cosmology units not turned on by default)
+        table = QTable.read(filename, **kwargs)
+
+    # Need to map the table's column names to Cosmology inputs.
+    table_column_names = set(table.columns)
+    for cosmo_name, html_name in _FORMAT_TABLE.items():
+        if html_name in table_column_names:
+            table.rename_column(html_name, cosmo_name)
+
+    # Build the cosmology from table, using the private backend.
+    return from_table(table, index=index, move_to_meta=move_to_meta, cosmology=cosmology)
+
+
+def write_html_table(cosmology: Cosmology, file, *, overwrite=False, cls=QTable, **kwargs):
+    """Serialize the |Cosmology| into a HTML table.
+
+    Parameters
+    ----------
+    cosmology : |Cosmology| subclass instance file : path-like or file-like
+        Location to save the serialized cosmology.
+    file : path-like or file-like
+        Where to write the html table.
+
+    overwrite : bool, optional keyword-only
+        Whether to overwrite the file, if it exists.
+    cls : |Table| class, optional keyword-only
+        Astropy |Table| (sub)class to use when writing. Default is |QTable|
+        class.
+    **kwargs : Any
+        Passed to ``cls.write``.
+
+    Raises
+    ------
+    TypeError
+        If the optional keyword-argument 'cls' is not a subclass of |Table|.
+
+    Examples
+    --------
+    A HTML file containing a Cosmology HTML table should have scripts enabling
+    MathJax.
+
+    ::
+        <script
+        src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
+        <script type="text/javascript" id="MathJax-script" async
+            src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml.js">
+        </script>
+    """
+    # Set cosmology_in_meta as false for now since there is no metadata being kept
+    table = to_table(cosmology, cls=cls, cosmology_in_meta=False)
+
+    cosmo_cls = type(cosmology)
+    n: str
+    for n, col in zip(table.colnames, table.itercols()):
+        param = getattr(cosmo_cls, n, None)
+        if not isinstance(param, Parameter) or param.unit in (None, u.one):
+            continue
+        # Replace column with unitless version
+        table.replace_column(n, (col << param.unit).value, copy=False)
+
+    for n in cosmology.__parameters__:
+        new_name = _FORMAT_TABLE.get(n, n)
+        table.rename_column(n, new_name)
+
+    # Write HTML, using table I/O
+    kwargs["format"] = "ascii.html"  # force the format

```

Comment:
I don't think one should ignore possibly wrong `format` input, so I'd suggest either using  `kwargs.setdefault("format", "ascii.html")`, i.e., if the user wants to screw this up, let them, or pass on `format="ascii.html"` in the actual call and get an error if `kwargs` contains `format` too.